### DISCUSSION TITLE
Tuya Cloud fhempy

### DISCUSSION THREAD

**Discussion by dsossna** [fhempy/fhempy#92] [General]
Hello everybody,
I installed fhempy and tryed to install tuya cloud connector. After an update I get this error message from the connector module.

Failed to load module tuya_cloud: Traceback (most recent call last): File "/opt/fhem/.local/lib/python3.7/site-packages/fhempy/lib/fhem_pythonbinding.py", line 318, in handle_function module_object = await self.import_module(hash) File "/opt/fhem/.local/lib/python3.7/site-packages/fhempy/lib/fhem_pythonbinding.py", line 526, in import_module functools.partial(importlib.import_module, pymodule) File "/opt/fhem/.local/lib/python3.7/site-packages/fhempy/lib/utils.py", line 35, in run_blocking return await asyncio.get_event_loop().run_in_executor(pool, function) File "/usr/lib/python3.7/concurrent/futures/thread.py", line 57, in run result = self.fn(*self.args, **self.kwargs) File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module return _bootstrap._gcd_import(name[level:], package, level) File "", line 1006, in _gcd_import File "", line 983, in _find_and_load File "", line 967, in _find_and_load_unlocked File "", line 677, in _load_unlocked File "", line 728, in exec_module File "", line 219, in _call_with_frames_removed File "/opt/fhem/.local/lib/python3.7/site-packages/fhempy/lib/tuya_cloud/tuya_cloud.py", line 8, in import fhempy.lib.tuya_cloud.tuya_cloud_setup as tcs File "/opt/fhem/.local/lib/python3.7/site-packages/fhempy/lib/tuya_cloud/tuya_cloud_setup.py", line 5, in import tuya_iot File "/opt/fhem/.local/lib/python3.7/site-packages/tuya_iot/__init__.py", line 2, in from .device import TuyaDevice, TuyaDeviceListener, TuyaDeviceManager File "/opt/fhem/.local/lib/python3.7/site-packages/tuya_iot/device.py", line 7, in from typing import Any, Literal, Optional ImportError: cannot import name 'Literal' from 'typing' (/usr/lib/python3.7/typing.py)

The failed to load module ?

regards Dirk

**Comment by fhempy** [comment]
Hi,
tuya_cloud requires Python 3.8 or higher.

**Reply by dsossna** [reply]
Thank you

**Reply by dsossna** [reply]
now I get this message: 'code': 1004, 'msg': 'sign invalid', 'success': False, 't':

**Reply by fhempy** [reply]
Invalid credentials or iot project at TuYa cloud too old?
